### PR TITLE
Animation position should be calculated correctly if target is svg element (T891214)

### PR DESCRIPTION
**Pull request by ksercs**

### CODE REVIEW

File: js/core/utils/position.js [hunk 1]
```diff
@@ -1,9 +1,36 @@
...
 import config from '../config';
+import typeUtils from '../utils/type';
 
 const getDefaultAlignment = function(isRtlEnabled) {
     const rtlEnabled = isRtlEnabled ?? config().rtlEnabled;
 
     return rtlEnabled ? 'right' : 'left';
 };
 
+const getBoundingRect = (instance) => {
```
Comment (San4es) [inline js/core/utils/position.js]: - Do we need yet another [size utils](https://github.com/DevExpress/DevExtreme/blob/4a46c3a58e8275d4c3933506efb07c5fb4795334/js/core/utils/size.js#L37)? 
- If yes, do we need to use it [everywhere](https://github.com/DevExpress/DevExtreme/search?q=getBoundingClientRect&unscoped_q=getBoundingClientRect)?
- Can we use our [`outerWidth`](https://github.com/DevExpress/DevExtreme/blob/4a46c3a58e8275d4c3933506efb07c5fb4795334/js/core/renderer_base.js#L167) implementation?

File: testing/tests/DevExpress.ui.widgets/popover.tests.js [hunk 2]
```diff
@@ -636,6 +636,74 @@ QUnit.module('content positioning', () => {
         }
     });
 
+    QUnit.test('Popover should be positioned correctly on the target right side when target is inside of svg (T891214)', function(assert) {
```
Comment (San4es) [inline testing/tests/DevExpress.ui.widgets/popover.tests.js:639]: What about adding tests for the `position` utils (see [position.tests.js](https://github.com/DevExpress/DevExtreme/blob/20_2/testing/tests/DevExpress.animation/position.tests.js))?

File: js/core/utils/position.js (hunk 1 above)
Comment (ksercs) [reply js/core/utils/position.js]: 1) No, this size util is too specific
2) Yeah, I gonna do it
3) No, it's not fully clear how to do it from a technical point of view

File: testing/tests/DevExpress.ui.widgets/popover.tests.js (hunk 2 above)
Comment (ksercs) [reply testing/tests/DevExpress.ui.widgets/popover.tests.js:639]: This module tests common position utils, etc calculatePosition - it's to common for adding a test for my new code. Besides, my new util is used in the main calculatePosition function, so it affects all of these tests

**Review by DokaRus — DISMISSED** [review]

File: testing/tests/DevExpress.ui.widgets/popover.tests.js (hunk 2 above)
Comment (ksercs) [reply testing/tests/DevExpress.ui.widgets/popover.tests.js:639]: I've add test to another test module, see the changes

**Review by DokaRus — DISMISSED** [review]
LGTM

File: js/core/utils/position.js [hunk 3]
```diff
@@ -1,9 +1,41 @@
...
+    const element = $(instance).get(0);
```
Comment (nightskylark) [inline js/core/utils/position.js]: I don't think it's a good idea to use renderer in the utility. It enlarges coupling. I'm sure that utility that works with DOM nodes do not need to know about the `renderer` entity. Maybe only DOM nodes should be allowed as an entry argument for this method?

Comment (ksercs) [reply js/core/utils/position.js]: I thought about it. I agree that coupling  enlarges, but also util external api becomes more flexible. I decided that the second aspect is more important. But ok, i'll fix it

File: js/core/utils/position.js [hunk 4]
```diff
@@ -1,9 +1,40 @@
...
+import typeUtils from '../utils/type';
 
 const getDefaultAlignment = function(isRtlEnabled) {
     const rtlEnabled = isRtlEnabled ?? config().rtlEnabled;
 
     return rtlEnabled ? 'right' : 'left';
 };
 
+const getBoundingRect = (instance) => {
+    const element = instance.get && instance.get(0) || instance;
```
Comment (San4es) [inline js/core/utils/position.js]:
```suggestion
const getBoundingRect = (element) => {
    element = element.get && element.get(0) || element;
```

File: js/core/utils/position.js [hunk 5]
```diff
@@ -1,9 +1,40 @@
...
+            width: element.outerWidth,
+            height: element.outerHeight
+        };
+    }
+
+    let rect;
+    try {
+        rect = element.getBoundingClientRect();
+    } catch(e) {
+        // TODO: add warning "IE throws Unspecified error if there is no such element on the page DOM"
```
Comment (San4es) [inline js/core/utils/position.js]: When we are going to add this warning?

**Review by nightskylark — DISMISSED** [review]

File: js/core/utils/position.js (hunk 5 above)
Comment (ksercs) [reply js/core/utils/position.js]: Probably, we will stop supporting the IE, and then the need for warning will disappear. I don’t think it’s worth doing it now - it's, at least, additional work with the documentation. And the scenario is rare. (The mere fact that the IE does not describe this error makes you think)

**Review by DokaRus — DISMISSED** [review]

File: js/core/utils/position.js (hunk 5 above)
Comment (San4es) [reply js/core/utils/position.js]:
```suggestion
        // NOTE: IE throws 'Unspecified error' if there is no such element on the page DOM
```

**Review by San4es — APPROVED** [review]

**Review by DokaRus — APPROVED** [review]

**Pull request merged**